internal/api/api.go

```diff
@@ -297,36 +263,47 @@ func routeRoomCommandUseradd(w http.ResponseWriter, req *http.Request) {
 
 	fingerprint := string(content)
 
-	if err := daemon.AddUserToRoom(roomID, fingerprint); err != nil {
+	if err := daemon.AddPeerToRoom(roomID, fingerprint); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
 
-func routeRoomCommandNameroom(w http.ResponseWriter, req *http.Request) {
-	content, err := ioutil.ReadAll(req.Body)
+func RouteRoomCommandNameRoom(w http.ResponseWriter, req *http.Request) {
+	err, errCode := sendMessage(req, types.RoomCommandNameRoom)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		http.Error(w, err.Error(), errCode)
 	}
+}
 
-	msg := "name_room " + string(content)
-	err = daemon.SendMessage(req.FormValue("uuid"), types.MessageTypeCmd, []byte(msg))
+func RouteRoomCommandSetNick(w http.ResponseWriter, req *http.Request) {
+	err, errCode := sendMessage(req, types.RoomCommandNick)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		http.Error(w, err.Error(), errCode)
 	}
 }
 
-func routeRoomCommandSetnick(w http.ResponseWriter, req *http.Request) {
+func sendMessage(req *http.Request, roomCommand types.RoomCommand) (error, int) {
```

Comment:
[Deepsoruce](https://deepsource.io/gh/craumix/onionmsg/run/bc903e67-7016-4b2b-8d77-f936228a3597/go/SCC-ST1008)
Hast du einen Grund für die Reihenfolge?

Reply:
http.Error hat sie in der Reihenfolge